src/item.cpp

```diff
@@ -2978,6 +3013,34 @@ int item::chip_resistance( bool worst ) const
     return res;
 }
 
+int item::damage_resist( damage_type dt, bool to_self ) const
+{
+    switch( dt ) {
+        case DT_TRUE:
+        case DT_BIOLOGICAL:
+        case DT_ELECTRIC:
+        case DT_COLD:
+            // Currently hardcoded:
+            // Items can never be damaged by those types
+            // But they provide 0 protection from them
+            return to_self ? INT_MAX : 0;
+        case DT_BASH:
+            return bash_resist( to_self );
+        case DT_CUT:
+            return cut_resist ( to_self );
+        case DT_ACID:
+            return acid_resist( to_self );
+        case DT_STAB:
+            return stab_resist( to_self );
+        case DT_HEAT:
+            return fire_resist( to_self );
+        default:
```

Comment:
Why don't you just leave out the default case and let the compiler warn you whenever a new enum value gets added?